File: graphql-subscriptions-apollo-express/graphql/resolvers/comments/create_comment.js
const {Comment} = require('../../../database/models');
const {pubsub} = require('../../../pubsub/local_pubsub');

const createComment = async (_, {postId, body}) => {
    const comment = await Comment.create({post_id: postId, body: body});
    pubsub.publish('commentAdded', {commentAdded: comment, postId: comment.post_id});
    return comment;
};

module.exports = createComment;

File: graphql-subscriptions-rails/app/graphql/mutations/create_comment.rb
# frozen_string_literal: true

module Mutations
  class CreateComment < Mutations::MutationSupport
    def resolve(_object, args, _context)
      post = Post.find(args[:postId])
      comment = Comment.create! do |comment|
        comment.post = post
        comment.body = args[:body]
      end
      MySchema.subscriptions.trigger('commentAdded', {}, comment)
      comment
    end
  end
end
